docs/source/dev/test.rst

```diff
@@ -0,0 +1,107 @@
+.. highlight:: cpp
+
+Writing a unit test
+==================
+
+After implementing a new functionality in Alpaka, it is recommended to test it. Indeed, having no compile time errors
+when building alpaka with any backend doesn't mean that such functionality is well implemented and behaves as expected.
+
+Unit tests are written and integrated with `Catch2`_ and they are standalone executables located in the ``test/unit`` and
+``test/integ`` paths inside the ``build`` folder. They can be built all at once by setting the ``CMake``
+argument ``-DBUILD_TESTING=ON`` and running the command ``cmake --build .``, and then run with the ``ctest`` command as
+described in the :doc:`Installation section </basic/install>`.
+
+.. _Catch2: https://github.com/catchorg/Catch2
+
+In alternative, a target test can be built by adding the flag ``--target <testname>`` to the ``cmake --build`` command and
+then executed as a normal binary file (``./<testname>``) in the corresponding folder mentioned above.
+
+Sample workflow
+-------
+
+As a sample workflow, consider the ``getWarpSize`` functionality that, given an ``alpaka device``, returns the number
+of threads that constitute a warp.
+
+First of all, the unit test should be located inside a folder under the ``test/unit`` path (namely ``dev``). Such a folder
+will contain a ``CMakeLists.txt`` file and a ``src`` folder, in which the ``.cpp`` file with the test itself will be located.
+
+A simple ``CMakeLists.txt`` file will look like:
+
+.. code-block::
+    set(_TARGET_NAME "devTest")
+
+    append_recursive_files_add_to_src_group("src/" "src/" "cpp" _FILES_SOURCE)
+
+    alpaka_add_executable(
+        ${_TARGET_NAME}
+        ${_FILES_SOURCE})
+    target_link_libraries(
+        ${_TARGET_NAME}
+        PRIVATE common)
+
+    set_target_properties(${_TARGET_NAME} PROPERTIES FOLDER "test/unit")
+
+    add_test(NAME ${_TARGET_NAME} COMMAND ${_TARGET_NAME} ${_ALPAKA_TEST_OPTIONS})
```

Comment:
I am hesitant to show the content of a `CMakeLists.txt` because it does not add a lot of value and might change at some point. And users are free to implement their `CMakeLists.txt` differently.